cli/commands/service.go

```diff
@@ -0,0 +1,142 @@
+		if err != nil {
+			return nil, err
+		}
+		return resolvedOptionsBytes, nil
+	}
+	return nil, nil
+}
+
+func (cmd *DescribeHandler) DescribeConfiguration(c *kingpin.ParseContext) error {
+	requestContent, _ := json.Marshal(DescribeRequest{config.ServiceName})
+	response := client.HTTPCosmosPostJSON("describe", string(requestContent))
+	responseBytes := client.GetResponseBytes(response)
+	resolvedOptionsBytes, err := parseDescribeResponse(responseBytes)
+	if err != nil {
+		reportErrorAndExit(err, responseBytes)
+	}
+	if resolvedOptionsBytes != nil {
+		client.PrintJSONBytes(resolvedOptionsBytes, nil)
+	} else {
+		log.Printf("No user options stored for service %s.", config.ServiceName)
+		log.Fatalf("User options are only persisted for packages installed with Enterprise DC/OS 1.10 or newer.")
+	}
+	return nil
+}
+
+type UpdateHandler struct {
+	UpdateName     string
+	OptionsFile    string
+	PackageVersion string
+	Status         bool
+}
+
+type UpdateRequest struct {
+	AppID          string                 `json:"appId"`
+	PackageVersion string                 `json:"packageVersion,omitempty"`
+	OptionsJSON    map[string]interface{} `json:"options,omitempty"`
+}
+
+func printStatus() {
+	log.Printf("Status has not been implemented yet. Please use `dcos %s --name=%s plan show` to view progress.", config.ModuleName, config.ServiceName)
+}
+
+func readJSONFile(filename string) (map[string]interface{}, error) {
+	fileBytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	return client.UnmarshalJSON(fileBytes)
+}
+
+func parseUpdateResponse(responseBytes []byte) (string, error) {
+	responseJSON, err := client.UnmarshalJSON(responseBytes)
+	if err != nil {
+		return "", err
+	}
+	return string(responseJSON["marathonDeploymentId"].(string)), nil
+}
+
+func doUpdate(optionsFile, packageVersion string) {
+	request := UpdateRequest{AppID: config.ServiceName}
+	if len(packageVersion) > 0 {
+		request.PackageVersion = packageVersion
+	}
+	if len(optionsFile) > 0 {
+		optionsJSON, err := readJSONFile(optionsFile)
+		if err != nil {
+			log.Fatalf("Failed to load specified options file %s: %s", optionsFile, err)
+		}
+		request.OptionsJSON = optionsJSON
+	}
+	requestContent, _ := json.Marshal(request)
+	response := client.HTTPCosmosPostJSON("update", string(requestContent))
+	responseBytes := client.GetResponseBytes(response)
+	// TODO: do something interesting with update response
+	_, err := parseUpdateResponse(responseBytes)
+	if err != nil {
+		reportErrorAndExit(err, responseBytes)
+	}
+	client.PrintText(fmt.Sprintf("Update started. Please use `dcos %s --name=%s service update --status` to view progress.", config.ModuleName, config.ServiceName))
+}
+
+func (cmd *UpdateHandler) UpdateConfiguration(c *kingpin.ParseContext) error {
+	if cmd.Status {
+		printStatus()
+		return nil
+	}
+	doUpdate(cmd.OptionsFile, cmd.PackageVersion)
+	return nil
+}
+
+func HandleServiceSection(app *kingpin.Application) {
```

Comment:
Done! Note that this breaks the one "section" per file convention that we've been going with. I kept them in the same file so that the code for these commands and their related code are grouped logically.